tutormfe/plugin.py

```diff
@@ -43,22 +42,66 @@
     },
 }
 
-hooks = {
-    "build-image": {
-        "mfe": "{{ MFE_DOCKER_IMAGE }}",
-    },
-    "remote-image": {
-        "mfe": "{{ MFE_DOCKER_IMAGE }}",
-    },
-    "init": ["lms"],
-}
+hooks.Filters.COMMANDS_INIT.add_item(
+    (
+        "lms",
+        ("mfe", "tasks", "myservice", "init"),
+    )
+)
+hooks.Filters.IMAGES_BUILD.add_item(
+    (
+        "mfe",
+        ("plugins", "mfe", "build", "mfe"),
+        "{{ MFE_DOCKER_IMAGE }}",
+        (),
+    )
+)
+
+
+@hooks.Filters.IMAGES_PULL.add()
+@hooks.Filters.IMAGES_PUSH.add()
+def _add_remote_mfe_image_iff_customized(images, user_config):
+    """
+    Register MFE image for pushing & pulling if and only if it has
+    been set to something other than the default.
+
+    This is work-around to an upstream issue with MFE config. Briefly:
+    User config is baked into MFE builds, so Tutor cannot host a generic
+    pre-built MFE image. Howevever, individual Tutor users may want/need to
+    build and host their own MFE image. So, as a compromise, we tell Tutor
+    to push/pull the MFE image if the user has customized it to anything
+    other than the default image URL.
+    """
+    image_tag = user_config["MFE_DOCKER_IMAGE"]
+    if not image_tag.startswith("docker.io/overhangio/openedx-mfe:"):
+        # Image has been customized. Add to list for pulling/pushing.
+        images.append(("mfe", image_tag))
+    return images
 
+####### Boilerplate code
```

Comment:
Consider adding some more detail to this comment. The boilerplate code could be somewhat unapproachable to someone unfamiliar with the new plugin API. For example, `# Render all templates under tutormfe/templates`, `# Load all files under tutormfe/patches as patches`, etc